Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/lucene/impl/LuceneUtility.java

```diff
@@ -622,7 +622,9 @@ private void buildQuery(BooleanClause.Occur occur, PerFieldAnalyzerWrapper perFi
 
     private void buildTokenizedOrWildCardQuery(LuceneField luceneFieldAnnotation, String searchString,
             BooleanQuery.Builder tokenizedQueryBuilder) {
-        for (String searchValue : searchString.split(" ")) {
+        String hypSearchStr = new String(searchString);
+        hypSearchStr = searchString.replace('-', ' ');
+        for (String searchValue : hypSearchStr.split(" ")) {
             if (luceneFieldAnnotation.isWildCardSearchEnabled()) {
                 createWildCardSearchQuery(luceneFieldAnnotation, searchValue, tokenizedQueryBuilder, Occur.MUST);
```

Comment:
Does the value of luceneFieldAnnotation.isWildCardSearchEnabled() change for every iteration? 

If not, its better to write the if-else condition and then include the iteration call inside them: 
String splitHypSearchStr =  hypSearchStr.split(" ");
if(luceneFieldAnnotation.isWildCardSearchEnabled()){
     // iterate splitHypSearchStr and call the method createWildCardSearchQuery
}
else{
  // iiterate splitHypSearchStr and call the method tokenizedQueryBuilder.add
}

Reply:
The value of luceneFieldAnnotation.isWildCardSearchEnabled() doesn't change for every iteration of hypSearchStr. But instead of having 2 iterators isn't it better to have just one iterator outside the if else?

Reply:
Given that the value of  luceneFieldAnnotation.isWildCardSearchEnabled()  doesn't change for every iteration, iI feel its good to have the iterations inside both if and else loop, instead of checking the conditions (if/else) for every iteration.  If you feel that the code will look cluttered due to the for-loops inside if and else, consider : 
1) 
https://stackoverflow.com/questions/27534684/good-method-to-call-method-on-each-object-using-stream-api
2) 
https://stackoverflow.com/questions/36508530/converting-array-iteration-to-lambda-expression